org.scala-ide.sdt.core/src/org/scalaide/core/internal/project/scopes/BuildScopeUnit.scala

```diff
@@ -0,0 +1,82 @@
+package org.scalaide.core.internal.project.scopes
+
+import scala.tools.nsc.Settings
+import org.eclipse.core.resources.IContainer
+import org.eclipse.core.resources.IFile
+import org.eclipse.core.resources.IResource
+import org.eclipse.core.runtime.IProgressMonitor
+import org.eclipse.core.runtime.SubMonitor
+import org.eclipse.jdt.core.IJavaModelMarker
+import org.scalaide.core.IScalaProject
+import org.scalaide.core.SdtConstants
+import org.scalaide.core.internal.builder.BuildProblemMarker
+import org.scalaide.core.internal.builder.zinc.SbtAwareBuildManager
+import org.scalaide.core.internal.project.CompileScope
+import sbt.inc.IncOptions
+import org.scalaide.core.internal.builder.zinc.EclipseSbtBuildManager
+
+class BuildScopeUnit(val scope: CompileScope, val owningProject: IScalaProject, settings: Settings,
+  val dependentUnitInstances: Seq[BuildScopeUnit] = Seq.empty)
+    extends SbtAwareBuildManager {
+
+  private val delegate =
+    new EclipseSbtBuildManager(owningProject, settings, Some(owningProject.underlying.getFile(".cache-" + scope.name)),
+        addToClasspath, srcOutputs)
+
+  private def isSrcOutPossessor: PartialFunction[(IContainer, IContainer), Boolean] = {
+    case (src, out) if scope.isValidSourcePath(src.getProjectRelativePath) => true
+    case _ => false
+  }
+
+  private def addToClasspath = owningProject.sourceOutputFolders.filterNot(isSrcOutPossessor) map {
```

Comment:
`additionalEntries`? `dependentOutputFolders`? Should this contain only the output folders that are "above" in the compilation pipeline? for instance, for `Main` it should only contain output folders from `Macros`, not `Test`. After a successful build, `Main` will "see" test classes from the previous build, no?